--- noip-2.1.9/Makefile.orig	2008-11-21 23:27:15.000000000 +0100
+++ noip-2.1.9/Makefile	2008-11-25 08:36:44.392738218 +0100
@@ -1,10 +1,12 @@
-TGT=noip2
+TGT=noip
 CC=gcc
+CFLAGS="-Wall -g -O2"
 PKG=noip-2.1.tgz
 
 PREFIX=/usr/local
 CONFDIR=${PREFIX}/etc
-BINDIR=${PREFIX}/bin
+SBINDIR=${PREFIX}/sbin
+DESTDIR=
 
 # these defines are for Linux
 LIBS=
@@ -22,14 +24,14 @@
 # ARCH=sun
 
 ${TGT}: Makefile ${TGT}.c 
-	${CC} -Wall -g -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
+	${CC} $(CFLAGS) -D${ARCH} -DPREFIX=\"${PREFIX}\" \
+	-DCONFIG_FILENAME=\"$(CONFDIR)/noip.conf\" ${TGT}.c -o ${TGT} ${LIBS}
 
 install: ${TGT} 
-	if [ ! -d ${BINDIR} ]; then mkdir -p ${BINDIR};fi
-	if [ ! -d ${CONFDIR} ]; then mkdir -p ${CONFDIR};fi
-	cp ${TGT} ${BINDIR}/${TGT}
-	${BINDIR}/${TGT} -C -c /tmp/no-ip2.conf
-	mv /tmp/no-ip2.conf ${CONFDIR}/no-ip2.conf
+	if [ ! -d $(DESTDIR)$(SBINDIR) ]; then install -d $(DESTDIR)$(SBINDIR);fi
+	if [ ! -d $$(DESTDIR)$(CONFDIR) ]; then install -d $(DESTDIR)$(CONFDIR);fi
+	if [ ! -d $(DESTDIR)/tmp ]; then install -d $(DESTDIR)/tmp;fi
+	cp ${TGT} $(DESTDIR)$(SBINDIR)/${TGT}
 
 package: ${TGT}
 	rm  -f *.bak
